server/channels/app/bot.go

```diff
@@ -124,7 +124,9 @@ func (a *App) CreateBot(rctx request.CTX, bot *model.Bot) (*model.Bot, *model.Ap
 
 	savedBot, nErr := a.Srv().Store().Bot().Save(bot)
 	if nErr != nil {
-		a.Srv().Store().User().PermanentDelete(rctx, bot.UserId)
+		if err := a.Srv().Store().User().PermanentDelete(rctx, bot.UserId); err != nil {
+			a.Log().Error("Failed to permanently delete user", mlog.Err(err))
```

Comment:
I wonder if the error message could be more descriptive. It's currently not clear, why the code deletes the users.

Reply:
How about this:
> "Failed to permanently delete the user after bot save failure"